src/basic/process-util.c

```diff
@@ -1632,14 +1686,19 @@ int safe_fork_full(
                                 return -EPROTO;
                 }
 
-                if (ret_pid)
-                        *ret_pid = pid;
+                if (ret_pid) {
+                        r = pidref_set_pid(ret_pid, pid);
```

Comment:
In the case of FORK_WAIT, the parent should always get PIDREF_NULL, since the process has exited